docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +278,229 @@ Note that:
 * Since the type of configuration file is used as an MQTT topic name, the characters `#` and `+` cannot be used in a type name.
   If such a character is used in a type name (or in the path of a configuration file without explicit type),
   then the whole plugin configuration `/etc/tedge/c8y/c8y-configuration-plugin.toml` is considered ill-formed.
+
+## Configuration protocol between thin-edge and the child-devices
+
+The configuration plugin `c8y_configuration_plugin` can act as a proxy between the cloud and a child-device.
+However, for that to work, a client must be installed on the child device
+to perform the actual configuration updates pushed by thin-edge on behalf of the cloud.
+While the configuration plugin tells what need to be updated and when,
+only the child device specific client can control where and how these updates can be applied.
+
+* The responsibility of the configuration plugin is to
+  * interact with the cloud, receiving the configuration updates and requests,
+  * publish and collect configuration files via an HTTP-based file transfer service over the local network,
+  * notify the child devices via MQTT when configuration files are to be updated or requested from the cloud,
+  * listen for child devices MQTT configuration status messages.
+* The child-device specific software is an MQTT+HTTP client that
+  * interact with the child-device system, updating and requesting the target configuration files
+  * connect the main thin-edge device over the local MQTT bus,
+  * listen over MQTT for configuration updates and requests,
+  * download and upload on demand the configuration files,
+  * notify the progress of the configuration updates to the main device via MQTT.
+
+For each kind of child device such a specific client has to be implemented
+and installed on the child-device hardware (To be precise, most often on the child device,
+but not necessarily as, for a child device which software cannot be altered,
+one might have to install a protocol adaptor on the main device).
```

Comment:
> (To be precise, most often on the child device,
but not necessarily as, for a child device which software cannot be altered,
one might have to install a protocol adaptor on the main device)

- Very long sentence and many commas
- Too heavy topic to mention it just as a side node in brackets

Reply:
I will consider to remove this. Or to add a picture. It's not clear for all where child device specific software is running.